tests/base.py

```diff
@@ -0,0 +1,1034 @@
+"""
+Setup expectations for test sub classes
+Run discovery for as a prerequisite for most tests
+"""
+import unittest
+import os
+import singer
+
+from datetime import timedelta
+from datetime import datetime as dt
+from datetime import timezone as tz
+
+from tap_tester import connections, menagerie, runner
+
+
+class SalesforceBaseTest(unittest.TestCase):
+    """
+    Setup expectations for test sub classes
+    Run discovery for as a prerequisite for most tests
+    """
+    AUTOMATIC_FIELDS = "automatic"
+    REPLICATION_KEYS = "valid-replication-keys"
+    PRIMARY_KEYS = "table-key-properties"
+    FOREIGN_KEYS = "table-foreign-key-properties"
+    REPLICATION_METHOD = "forced-replication-method"
+    API_LIMIT = "max-row-limit"
+    INCREMENTAL = "INCREMENTAL"
+    FULL_TABLE = "FULL_TABLE"
+    START_DATE_FORMAT = "%Y-%m-%dT00:00:00Z"
+
+    LOGGER = singer.get_logger()
+
+    @staticmethod
+    def tap_name():
+        """The name of the tap"""
+        return "tap-salesforce"
+
+    @staticmethod
+    def get_type():
+        """the expected url route ending"""
+        return "platform.salesforce"
+
+    def get_properties(self, original: bool = True):
+        """Configuration properties required for the tap."""
+        return {
```

Comment:
Am I missing something or does this return mean all the code after here is not getting executed? I would ask how pylint doesn't catch this, but I know we don't normally run pylint on the test repo (which I think is wrong)... But yeah, did you intend to return here and not run the rest of the code in this method?

Reply:
Nope, that is a HUGE mistake. This could cause the start date test to give a false negative. I think that is the only test that uses this logic. Will address.

Reply:
Fixed